plugins/outputs/azure_monitor_logs/azure_monitor_logs.go

```diff
@@ -0,0 +1,253 @@
+package azure_monitor_logs
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+var sampleConfig = `
+  ## Customer ID (Workstation ID) and Key for Azure Log Analytics resource.
+  # customer_id = "<Workstation ID>"
+  # shared_key = "<Secret>"
+
+  ## Timeout for closing (default: 5s).
+  # timeout = "5s"
+
+  ## Table Namespace Prefix (default: "").
+  ## Namespace Prefix is used in "Log-Type" header
+  ## Prefix can only contain alphaNumeric characters
+  # namespace_prefix = ""
+`
+
+const (
+	baseURL                = "https://%s.ods.opinsights.azure.com/api/logs?api-version=2016-04-01"
+	contentType            = "application/json"
+	timeGeneratedFieldName = "DateTime"
+	namespacePrefixRegex   = "^[a-zA-Z0-9]*$"
+	httpMethod             = http.MethodPost
+
+	defaultClientTimeout = 5 * time.Second
```

Comment:
While being a matter of taste, I'm not a big fan of defining constants that are later used only once in the code and will _most likely_ never change. I leave it to you to keep those or move to the actual places where those are used (once).

Reply:
I prefer to have those located at the top of the file.